h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -2291,6 +2302,18 @@ private static Value oraHash(Value value, long bucket, long seed) {
         return ValueBigint.get((hc & Long.MAX_VALUE) % (bucket + 1));
     }
 
+    private static Value base64encode(byte[] data, String algorithm) {
+        final EncodingAlgorithm algorithmImpl = EncodingFactory.getEncodingAlgorithm(algorithm);
+        final String result = algorithmImpl.encode(data);
+        return ValueVarchar.get(result);
+    }
+
+    private static ValueVarbinary base64decode(String data, String algorithm) {
+        final EncodingAlgorithm algorithmImpl = EncodingFactory.getEncodingAlgorithm(algorithm);
+        final byte[] result = algorithmImpl.decode(data);
+        return ValueVarbinary.getNoCopy(result);
+    }
```

Comment:
Please, do the same here.

But this function is more specific. If `algorithm` is not specified, it's more reasonable to guess the alphabet from the encoded string.
```Java
Base64.Decoder decoder = data.indexOf('-') < 0 && data.indexOf('_') <= 0
    ? Base64.getMimeDecoder()
    : Base64.getUrlDecoder();
```
It's better to use the MIME decoder for non-URL encoded strings (at least when `algorithm` is not specified).

You also need to catch possible `IllegalArgumentException` from decoder and throw a normal `DbException` instead of it.